components/script/dom/globalscope.rs

```diff
@@ -3558,32 +3478,147 @@ impl GlobalScope {
+
+        // TODO Step 2. If scripting is disabled for settings, then set source to the empty string.
 
-        // Steps 4-10
-        rooted!(in(*GlobalScope::get_cx()) let mut rval = UndefinedValue());
-        _ = self.evaluate_script_on_global_with_result(
-            &script.code,
-            script.url.as_str(),
-            rval.handle_mut(),
+        // TODO Step 4. Set script's settings object to settings.
+        // TODO Step 7. Set script's muted errors to mutedErrors.
+
+        // TODO Step 9. Record classic script creation time given script and sourceURLForWindowScripts.
+
+        // Step 10. Let result be ParseScript(source, settings's realm, script).
+        compiled_script.set(compile_script(
+            cx,
+            &source,
+            url.as_str(),
             line_number,
-            script.fetch_options.clone(),
-            script.url.clone(),
-            can_gc,
             introduction_type,
-        );
+        ));
+
+        // Step 11. If result is a list of errors, then:
+        let record = if compiled_script.get().is_null() {
+            // Set script's parse error and its error to rethrow to result[0].
+            rooted!(in(*cx) let mut exception = UndefinedValue());
+
+            assert!(unsafe { JS_GetPendingException(*cx, exception.handle_mut()) });
+            unsafe { JS_ClearPendingException(*cx) };
+
+            Err(RethrowError::new(Heap::boxed(exception.get())))
+        } else {
+            Ok(NonNull::new(*compiled_script).expect("Can't be null"))
+        };
+
+        // Step 3. Let script be a new classic script that this algorithm will subsequently initialize.
+        // Step 5. Set script's base URL to baseURL.
+        // Step 6. Set script's fetch options to options.
+        // Step 12. Set script's record to result.
+        let mut script = ClassicScript {
+            record,
+            url,
+            fetch_options,
+            source: source_code,
+            external,
+            unminified_dir: self.unminified_js_dir(),
+        };
+        unminify_js(&mut script);
+
+        // Step 13. Return script.
+        script
+    }
+
+    /// <https://html.spec.whatwg.org/multipage/#run-a-classic-script>
+    #[expect(unsafe_code)]
+    pub(crate) fn run_a_classic_script(&self, script: ClassicScript, can_gc: CanGc) -> ErrorResult {
+        let cx = GlobalScope::get_cx();
+        // TODO Step 1. Let settings be the settings object of script.
+
+        // Step 2. Check if we can run script with settings. If this returns "do not run", then return NormalCompletion(empty).
+        if !self.can_run_script() {
+            return Ok(());
+        }
+
+        // TODO Step 3. Record classic script execution start time given script.
+
+        // Step 4. Prepare to run script given settings.
+        // Once dropped this will run "Clean up after running script" steps
+        let _aes = AutoEntryScript::new(self);
+
+        // Step 5 Let evaluationStatus be null.
```

Comment:
Ultra-nits: use dots after step numbers to make it consistent with the other comments
```suggestion
        // Step 5. Let evaluationStatus be null.
```